main.cpp

```diff
@@ -478,6 +495,27 @@ IntType& IntType::pow(const IntType& it)
     return powInternal(static_cast<int>(it));
 }
 
+IntType& IntType::apply(std::function<IntType&(int&)> func)
+{
+    if(func)
+        func(*value);
```

Comment:
the lambda returns a reference to `IntType&`. 
This function also returns a reference to `IntType&`. 
```
 2) One of the apply() functions should takes a std::function<> object as the function argument.
    the std::function<> object should return *this;
```
if `func` is valid, then return the result that `func` returns from this function. 
- [ ] revise